pkg/autodiscovery/autoconfig.go

```diff
@@ -77,7 +77,7 @@ type AutoConfig struct {
 }
 
 // NewAutoConfig creates an AutoConfig instance.
-func NewAutoConfig(scheduler scheduler.Scheduler) *AutoConfig {
+func NewAutoConfig(scheduler *scheduler.MetaScheduler) *AutoConfig {
```

Comment:
The `AddScheduler` and `RemoveScheduler` methods do not belong in to `scheduler.Scheduler` interface that defines the interface the collector, log-agent and clustercheck handler implement.

Now that we are adding these methods, we actually don't need to pass a pre-configured metascheduler to the AutoConfig object, the agent could just call `AddScheduler` 

@mfpierre with the new `AddScheduler`, does it still make sense to inject a MetaScheduler to `NewAutoConfig` ? Current setup code: https://github.com/DataDog/datadog-agent/blob/5770a764a5e2c13ee4b75ee3a840fc252b7586b9/cmd/agent/common/autoconfig.go#L34-L41

Reply:
When I introduced the `Scheduler` interface it was basically free to put the interface instead of the `Collector` in the autoconfig constructor, now that we have specific methods I think it's ok to go back to a concrete type and we'll make the work, if at some point, we need to pass the interface but I'll prefer to avoid premature optimization as we don't need this for now.